pilot/cmd/pilot-discovery/admission/controller.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package admission
+
+import (
+	"github.com/hashicorp/go-multierror"
+	"istio.io/istio/pilot/platform/kube"
+	"istio.io/istio/pilot/platform/kube/admit"
+)
+
+// NewController creates and runs an instance of the pilot admission controller with the given arguments.
+func NewController(kubeContig string, args admit.ControllerOptions) (*admit.AdmissionController, error) {
+	_, client, kuberr := kube.CreateInterface(kubeContig)
+	if kuberr != nil {
```

Comment:
Eh? This entire PR is for non kube setup right? What is this doing here? Or is it code moved from elsewhere?

Reply:
No, this PR is to enable all use cases ... not just non-kube.  I guess you're suggesting that `main.go` should not create the admission controller unless a k8s service registry is used?

Reply:
The admission webhook is a function of the api-server and not the service registry. For example, standalone api-server with Eureka service registry could be configured to invoke admission controller for server-side config validation. The only case where we shouldn't create the admission controller is for file-based watch.

Reply:
@ayj that makes sense ... let me clean this up.